OlinkAnalyze/R/Olink_anova.R

```diff
@@ -443,9 +460,25 @@ olink_anova_posthoc <- function(df,
     stop("All effect terms must be included in the variable argument or model formula.")
   }
 
-
+  # Stop if control assays have not been removed
+  if ("AssayType" %in% names(df)) {
+    if (any(df$AssayType != "assay")) {
+      ctrl_assays <- df |>
+        dplyr::filter(AssayType != "assay")
+      
+      stop(paste0('Control assays have not been removed from the dataset.\n  Assays with AssayType != "assay" should be excluded.\n  The following control assays were found:\n  ',
+                  paste(strwrap(toString(unique(ctrl_assays$Assay)), width = 80), collapse = "\n")))
+    }
+  } else if (any(str_detect(df$Assay, regex("control|ctrl", ignore_case = TRUE)))) {
```

Comment:
package name needs to be specified stringr::str_detect and stringr::regex()

Reply:
Done